src/refs/mutable.rs

```diff
@@ -19,7 +20,7 @@ use crate::borrow::AtomicBorrowMut;
 /// [`AtomicCell`]: struct.AtomicCell.html
 /// [`&T`]: https://doc.rust-lang.org/core/primitive.reference.html
 pub struct RefMut<'a, T: ?Sized> {
-    value: &'a mut T,
+    value: NonNull<T>,
```

Comment:
`NonNull<T>` is covariant over `T` but `RefMut` allows mutable access so it needs to be invariant over `T` to be sound, to accomplish this a `PhamtomData<&'a mut T>` field can be included. https://doc.rust-lang.org/reference/subtyping.html#variance